src/Pages/Profile/admin/AdminView.jsx

```diff
@@ -0,0 +1,355 @@
+            }}
+            onClick={() => { handleToggle() }}>
+            Edit
+          </Button>
+        </div>
+
+        <Modal isOpen={toggle}>
+          <ModalHeader>Modal title</ModalHeader>
+          <ModalBody>
+            <Form>
+
+              {FormGroups.map((group, index) => {
+                return <FormGroup key={index}>
+                  <Label>{group.label}</Label>
+                  <Input
+                    type={group.type}
+                    name={group.type}
+                    placeholder={group.placeholder}
+                    onChange={group.onChange}
+                  />
+                </FormGroup>
+              })}
+
+              {/*Need Improvements, not doing anything currently*/}
+              {/* <FormGroup>
+                <Label for="exampleFile">Upload New Profile Picture</Label>
+                <Input
+                  type="file"
+                  name="file"
+                  id="exampleFile"
+                  onChange={event => {
+                    change(event, "profile");
+                  }}
+                />
+                <FormText color="muted">(Not Working!)</FormText>
+              </FormGroup> */}
+
+              Change validation date to
+              <select onChange={(e)=>{setNumberOfSemestersToSignUpFor(e.target.value)}}>
+                <option value={0}>Keep Same</option>
+                <option value={1}>This semester</option>
+                <option value={2}>2 semesters</option>
+              </select>
+
+              <Button
+                type='button'
+                onClick={() => {
+                  setPagesPrinted(0);
+                }}
+                color="info"
+                style={{marginTop:'5px'}}>
+                Reset Pages!
+              </Button>
+
+              <FormGroup tag="fieldset">
+                <legend>Membership Status</legend>
+                {memberships.map((membership, index) => {
+                  return <FormGroup check key={index}>
+                    <Label check>
+                      <Input
+                        type="radio"
+                        name="radio1"
+                        value={membership}
+                        onChange={() => {
+                          setuserMembership(membership)
+                        }}
+                      />
+                      {roleTranslator(membership)}
+                    </Label>
+                  </FormGroup>
+                })}
+              </FormGroup>
+
+            </Form>
+          </ModalBody>
+
+          <ModalFooter>
+            <Button
+              color="primary"
+              onClick={() => { handleSubmissionToggle() }}>
+              Submit
+            </Button>
+            <Button
+              color="secondary"
+              onClick={() => { handleToggle() }}>
+              Cancel
+            </Button>
+          </ModalFooter>
+
+        </Modal>
+      </div>
+    );
+  }
+
+  function display() {
```

Comment:
this should be it's own component. each thing to be rendered should be sent as props.